php-frontend/src/main/java/org/sonar/php/metrics/CognitiveComplexityVisitor.java

```diff
@@ -66,6 +67,42 @@ public static CognitiveComplexity complexity(FunctionTree functionTree) {
     return cognitiveComplexityVisitor.complexity;
   }
 
+  public static int complexity(CompilationUnitTree cut) {
+    // Only explicitly visit functions, methods and function expressions.
+    // Rest of the compilation unit is based on CognitiveComplexityVisitor computation
+    class CompilationUnitVisitor extends CognitiveComplexityVisitor {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make this a "static" inner class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2694)

Reply:
Still a ugly FP that I can not reproduce...

Reply:
https://jira.sonarsource.com/browse/SONARJAVA-2550